### PR TITLE
Add File Transfer hooks to SDKTools.

### PR DESCRIPTION
**Pull request by KyleSanderson**
I've ported this from a pretty dated bug on BZ, not even sure if it still builds...

### CODE REVIEW

**Comment by WildCard65** [comment]
Is this change suppose to include changes to the inc file?

**Comment by psychonic** [comment]
It would be great if the hook was disabled when there are no forward listeners. That way, if interfaces change, the plugin(s) using it can just be disabled temporarily, rather than all of SDKTools to avoid crashing until a fix is available.

File: extensions/sdktools/hooks.cpp [hunk 1]
```diff
@@ -45,10 +46,22 @@ static bool PRCH_enabled = false;
 static bool PRCH_used = false;
 
 SH_DECL_MANUALHOOK2_void(PlayerRunCmdHook, 0, 0, 0, CUserCmd *, IMoveHelper *);
+SH_DECL_HOOK2(IBaseFileSystem, FileExists, SH_NOATTRIB, 0, bool, const char*, const char *);
+#if SOURCE_ENGINE >= SE_ALIENSWARM
+SH_DECL_HOOK3(INetChannel, SendFile, SH_NOATTRIB, 0, bool, const char *, unsigned int, bool);
+#else
+SH_DECL_HOOK2(INetChannel, SendFile, SH_NOATTRIB, 0, bool, const char *, unsigned int);
+#endif
```
Comment (psychonic) [inline extensions/sdktools/hooks.cpp]: I believe that L4D2 has the bool as well (but not Nuclear Dawn). I'll look into making sure that our SDKs are up to date there.

**Comment by KyleSanderson** [comment]
> It would be great if the hook was disabled when there are no forward listeners.

Extensions race against core and CForward, there's no clean way to do this. When I was looking at the code again, I too thought of the same thing. Adding callbacks to the IForward iface is my intention to reduce the complexity and opt other forwards into this, but that's another PR.

**Comment by KyleSanderson** [comment]
> It would be great if the hook was disabled when there are no forward listeners.

https://github.com/alliedmodders/sourcemod/pull/307

**Comment by psychonic** [comment]
I think that there would be value in putting this into a separate extension (can still be first-party). With the filesystem interface being a little "breaky" on some games, it would be nice to be able to easily disable this functionality by moving the file out without having to disable all of SDKTools.

**Comment by KyleSanderson** [comment]
I've added an include file, hooks are not used until a plugin asks for one.

**Comment by Drifter321** [comment]
:ship:

**Pull request merged**